src/main.cpp

```diff
@@ -23,69 +23,71 @@
 
 using namespace std;
 
-// Reads edge list with comma separator
-pair<vector<pair<int, int>>, int> read_input(string filename, char delimiter = ',') {
+enum class Operation { READ, ADD, DELETE };
+
+// Reads edge list with separator
+pair<vector<tuple<Operation, int, int>>, int> read_mixed_input(string filename, Operation defaultOp) {
```

Comment:
The name `read_mixed_input` could be a bit misleading in cases where we are just reading a core graph, which is never mixed.

Reply:
Just to clarify, the updates are the only ones that are mixed and it would be good to distinguish that in the code.

Reply:
Thus, just `read_input` would be a good option without the need to have separate functions.

Reply:
I renamed the function to `read_input`